### DISCUSSION TITLE
Color Sequence view background by structure color theme?

### DISCUSSION THREAD

**Discussion by rdirisio** [molstar/molstar#919] [Q&A]
Hello,

I have a custom color theme that I use on displaying my structures. I would like to color the residues in the `Sequence` viewer with this color scheme. Is there a way to do this at the moment? I see that Sequence has a `getBackgroundColor` function, which just takes a marker number and colors according to if something is highlighted/selected. Perhaps the best course of action is to override this, but how do I go about getting the current theme of my structure to enable this?

Any advice is appreciated, thanks!

**Comment by dsehnal** [comment]
- There is no straightforward way to get "current theme" as different structure components can have different themes. Having said that, you can subscribe to the state tree updates and check what the current theme of the "polymer" component is for example.
- You will indeed need to write your own version of the sequence view component to implement this logic.
- You can create a separate instance of the theme you want to use to color the residues. Then you can create a single instance of a `StructureElement.Location` for which you update the current structure/element to pass to the theme (the highlighting code of the sequence view should help here)
- If you feel like contributing to the project, a possible general solution would be to add an extra "theme" drop-down to the built in sequence viewer. You can get a list of available color themes from `PluginContext.themes` property (or something close to that, I am writing this from the top of my head on mobile)

**Reply by rdirisio** [reply]
> There is no straightforward way to get "current theme" as different structure components can have different themes. Having said that, you can subscribe to the state tree updates and check what the current theme of the "polymer" component is for example. You will indeed need to write your own version of the sequence view component to implement this logic.

Yes, I think this is exactly what I will do. Thanks for this suggestion. What would be the best way to do get the current theme of a polymer? I am having difficulty finding the right plugin syntax for it.

**Reply by dsehnal** [reply]
`plugin.managers.hierarchy` should help here. You should be able to use that or at least use similar code patterns to find what you need.
